diff --git a/git.go b/git.go
index b7c8b3c..2d09710 100644
--- a/git.go
+++ b/git.go
@@ -143,15 +143,6 @@ func initLibGit2() {
 	C.git_libgit2_init()
 	features := Features()
 
-	// Due to the multithreaded nature of Go and its interaction with
-	// calling C functions, we cannot work with a library that was not built
-	// with multi-threading support. The most likely outcome is a segfault
-	// or panic at an incomprehensible time, so let's make it easy by
-	// panicking right here.
-	if features&FeatureThreads == 0 {
-		panic("libgit2 was not built with threading support")
-	}
-
 	if features&FeatureHTTPS == 0 {
 		if err := registerManagedHTTP(); err != nil {
 			panic(err)
